components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericStorageEngine.kt

```diff
@@ -290,6 +291,11 @@ internal abstract class GenericStorageEngine<MetricType> : StorageEngine {
         editor?.apply()
     }
 
+    override fun clearUserLifetimeMetrics() {
+        userLifetimeStorage.edit().clear().apply()
+        dataStores[Lifetime.User.ordinal].clear()
```

Comment:
Should we also have a way to clear or reset the firstRunFlag()?  My worry is that once things like the client_id are cleared, it will be difficult to re-populate those values without somehow resetting the first-run flag.

Reply:
I kinda like the idea of using the client_id as a first run flag...  and we just don't set it if glean is disabled to keep it from regenerating.

Reply:
This is a very good point!

I think I like the idea of ditching the file based "first run detector".